lib/common/client/configuration/base.rb

```diff
@@ -92,23 +92,30 @@ def request_options
         def breakers_service
           return @service if defined?(@service)
 
+          @service = create_new_breakers_service(breakers_matcher, breakers_exception_handler)
+        end
+
+        def breakers_matcher
           base_uri = URI.parse(base_path)
-          matcher = proc do |request_env|
+          proc do |request_env|
             request_env.url.host == base_uri.host && request_env.url.port == base_uri.port &&
               request_env.url.path =~ /^#{base_uri.path}/
           end
+        end
 
-          exception_handler = proc do |exception|
+        def breakers_exception_handler
+          proc do |exception|
             if exception.is_a?(Common::Exceptions::BackendServiceException)
               (500..599).cover?(exception.response_values[:status])
             elsif exception.is_a?(Common::Client::Errors::HTTPError)
               (500..599).cover?(exception.status)
+            elsif exception.is_a?(Faraday::ClientError)
```

Comment:
Are we going to make a note to update this once we upgrade to Faraday 1.0?

Reply:
It also looks like we're covering all 500 errors here, are we only concerned with 502's and 503's?

Reply:
my bad.  I was seeing examples the problem with 502/503 but I did mean to include the whole 500-range.

Reply:
Faraday has an [upgrade guide for 1.0](https://github.com/lostisland/faraday/blob/master/UPGRADING.md#faraday-10) that makes it clear, but I figured a comment wouldn't hurt either. I got lead astray because the [faraday documentation for ClientError](https://www.rubydoc.info/gems/faraday/Faraday/ClientError) changed as of 1.0